components/base/kinematicbase/execution.go

```diff
@@ -130,6 +130,16 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputSteps ...[]r
 		// - move until we think we have finished the arc, then move on to the next step
 		// - update our CurrentInputs tracking where we are through the arc
 		// - Check where we are relative to where we think we are, and tweak velocities accordingly
+
+		// Check if this arc is shorter than our typical check time; if so just run that and do not course correct.
+		if step.durationSeconds < updateStepSeconds {
+			utils.SelectContextOrWait(ctx, (time.Duration(step.durationSeconds*1000) * time.Millisecond))
```

Comment:
Can you not just write `step.durationSeconds` here?  The units on a `Duration` shouldnt matter IIRC

Reply:
The units on a duration absolutely matter. A duration is an `int`, so attempting to pass a float e.g. 0.4 seconds will be truncated to zero.